### PR TITLE
Fix private package install docs

### LINKED ISSUE #1824: Update Install Python Packages from Private Sources
@jedcunningham recommended the following changes to the topic to [Install Python packages from private sources](https://docs.astronomer.io/astro/develop-project?tab=github#install-python-packages-from-private-sources):

- Use double quotes vs single quotes when adding to PATH

- Don’t keyscan on the fly, hard code the github key, better to point folks [here](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/githubs-ssh-key-fingerprints). keyscan comment is okay as well for other services.

- Need to move the USER root up - @pohek321 to confirm.

@jedcunningham also made the following suggestions:

- Are those labels really necessary? Just extra noise imo.

- Instead of using base and reimplementing worse versions of the [onbuild commands](https://github.com/astronomer/astro-runtime/blob/main/image/Dockerfile.onbuild), why not use the onbuild image and just install the 1 extra private package with pip at the end? Less fragile to upstream image changes too.

These changes originated with this [Slack conversation](https://astronomer.slack.com/archives/C01N2DM2ZS8/p1675196726836089).

### PR DESCRIPTION
**Pull request by jedcunningham**
Fixes #1824 

This does switch to using the onbuild image (instead of base and multiphase) as if folks need these instructions, they probably should stick to onbuild. This also significantly simplifies the example.

### CODE REVIEW

File: astro/develop-project.md [hunk 1]
```diff
@@ -423,75 +422,32 @@ This example assumes that the name of each of your Python packages is identical
...
+2. Add `openssh-client` and `git` to your `packages.txt` file.
 
-  If you currently use the default distribution of Astro Runtime, replace your existing image with its corresponding `-base` image. The `-base` distribution is built to be customizable and does not include default build logic. For more information on Astro Runtime distributions, see [Distributions](runtime-image-architecture.md#distribution).
-
-  :::
-
-3. After the `FROM` line specifying your Runtime image, add the following configuration:
+3. In your Dockerfile, add the following instructions:
 
     ```docker
```
Comment (jedcunningham) [inline astro/develop-project.md:429]: I removed this as I felt it was implied, but happy to add it back:
```suggestion
    ```docker
    FROM quay.io/astronomer/astro-runtime:7.2.0
```

**Review by jedcunningham — COMMENTED** [review]
I didn't test this with an actual private package, but it did exercise all of the tooling that's used, so pretty confident this will work.

File: astro/develop-project.md [hunk 2]
```diff
@@ -403,18 +403,17 @@ This setup assumes that each custom Python package is hosted within its own priv
 
 #### Step 1: Specify the private repository in your project
 
-To add a Python package from a private repository to your Astro project, specify the repository’s SSH URL in your project’s `requirements.txt` file. The URL should use the following format:
+To add a Python package from a private repository to your Astro project, specify the repository’s SSH URL in a new `private-requirements.txt` file. The URL should use the following format:
```
Comment (kwcanuck) [inline astro/develop-project.md:406]:
```suggestion
To add a Python package from a private repository to your Astro project, specify the Secure Shell (SSH) URL for the repository in a new `private-requirements.txt` file. Use the following format for the SSH URL:
```

File: astro/develop-project.md [hunk 3]
```diff
@@ -423,75 +422,32 @@ This example assumes that the name of each of your Python packages is identical
...
-  :::
+    - Adds the fingerprint for GitHub to known_hosts
+    - Copies you private-requirements.txt file into the image
+    - Install Python-level packages from your private repository as specified in your `private-requirements.txt` file. Note, this securely mounts your SSH key at build time. This ensures that the key itself is not stored in the resulting Docker image filesystem or metadata.
+    - Adds the user bin directory onto PATH
 
   :::info
 
-  If your repository is not hosted on GitHub, replace the domain in the `ssh-keyscan` command with the domain where the package is hosted.
+  If your repository is not hosted on GitHub, replace the fingerprint with one from where the package is hosted. `ssh-keyscan` can be used to generate the fingerprint, if you don't already know it.
```
Comment (kwcanuck) [inline astro/develop-project.md:447]:
```suggestion
  If your repository isn't hosted on GitHub, replace the fingerprint with the hosted package location. Use `ssh-keyscan` to generate the fingerprint,
```

**Comment by jwitz** [comment]
@jedcunningham Are you still actively working on this draft, or is this ready for us to fully review and publish?

**Comment by jedcunningham** [comment]
@jwitz, feel free to take it and run with it.

File: astro/develop-project.md [hunk 4]
```diff
@@ -423,75 +422,32 @@ This example assumes that the name of each of your Python packages is identical
...
-     docker run quay.io/astronomer/astro-runtime:<runtime-version>-base python --version
-     ```
-
-  Make sure to replace `<runtime-version>` with your own.
-
-  :::
+    - Adds the fingerprint for GitHub to known_hosts
+    - Copies you private-requirements.txt file into the image
+    - Install Python-level packages from your private repository as specified in your `private-requirements.txt` file. Note, this securely mounts your SSH key at build time. This ensures that the key itself is not stored in the resulting Docker image filesystem or metadata.
+    - Adds the user bin directory onto PATH
```
Comment (jwitz) [inline astro/develop-project.md:443]:
```suggestion
    - Add the fingerprint for GitHub to `known_hosts`
    - Copy your `private-requirements.txt` file into the image
    - Install Python-level packages from your private repository as specified in your `private-requirements.txt` file. Tthis securely mounts your SSH key at build time, ensuring that the key itself is not stored in the resulting Docker image filesystem or metadata.
    - Add the user bin directory onto `PATH`
```

**Review by jwitz — APPROVED** [review]
LGTM! Thanks @jedcunningham and apologies on the delay with getting this merged.

**Pull request merged**